### DISCUSSION TITLE
TPOT- docker containers always restarting

### DISCUSSION THREAD

**Discussion by ghost6-2** [telekom-security/tpotce#1038] [Q&A] [answered]
Hi there, 
My plan is to implement a honeypot on my company network, and since we dont use VMware instead we use Microsotft Hyper-V we cannot import the ISO image. So I downloaded a clean install of Deabian Buster and went to github to manually install the TPOT.
Now I'm locally testing on my PC which I have VMware, and I'm trying to install it manually like I would do it on Hyper-V.  But after installing it I have a strange issue. My docker containers are always restarting, really don't know why are they not working and running up all the time.
So my steps of installation are next:

```
su -
git clone https://github.com/telekom-security/tpotce
cd tpotce/iso/installer/
cp tpot.conf.dist tpot.conf
./install.sh --type=auto --conf=tpot.conf
```
and then went with:

`./update.sh -y `

Below are the Q/A from the Issues section:

- What version of the OS are you currently using `lsb_release -a` and `uname -a`? -**Debian Buster 10.11**
- What T-Pot version are you currently using? **Latest version from github**
- What edition (Standard, Nextgen, etc.) of T-Pot are you running? -**Standard**
- What architecture are you running on (i.e. hardware, cloud, VM, etc.)? -**VM on Vmware**
- Did you have any problems during the install? If yes, please attach `/install.log` `/install.err`. - **No**
- How long has your installation been running? - **20-30 minutes**
- Did you install upgrades, packages or use the update script? **Used the update script**
- Did you modify any scripts or configs? If yes, please attach the changes.- **No**
- Please provide a screenshot of `glances` and `htop`.
![htop](https://user-images.githubusercontent.com/36500152/157624768-5fc60032-905b-4c67-8437-871bf03201a4.png)
![glances](https://user-images.githubusercontent.com/36500152/157624826-ac48b256-d51d-4941-8b94-cfa347e42aa0.png)

- How much free disk space is available (`df -h`)?**256 GB**
- What is the current container status (`dps.sh`)?
**This is where my issue lies.**
![status1](https://user-images.githubusercontent.com/36500152/157625057-0fc6b45b-0fc2-4b7b-ac33-13aeca937a1b.png)
![status2](https://user-images.githubusercontent.com/36500152/157625064-e312ae25-b082-4520-b4d5-742981516e83.png)
![status3](https://user-images.githubusercontent.com/36500152/157625074-7ea58b6a-8f38-402e-bf4b-77430fe2d0ff.png)

- What is the status of the T-Pot service (`systemctl status tpot`)?
![systemctl](https://user-images.githubusercontent.com/36500152/157625180-c5e765fb-5f5b-480e-9c5e-a796510977b5.png)

- What ports are being occupied? Stop T-Pot `systemctl stop tpot` and run `netstat -tulpen`
![netstat](https://user-images.githubusercontent.com/36500152/157625207-83fc6068-ee14-4545-ac76-f3c5c39d6aff.png)

- If a single container shows as `DOWN` you can run `docker logs <container-name>` for the latest log entries

My VM configuration:

![VM](https://user-images.githubusercontent.com/36500152/157628136-8ce3dd0d-e2a9-404e-9ead-dc2efb9bebe9.png)

Can please anyone help?

**Comment by t3chn0m4g3** [comment]
Run `systemctl stop tpot`.
Now test with `docker-compose -f /opt/tpot/etc/tpot.yml up` to see the log messages. If the containers are starting up and suddenly are breaking down, it might be a port conflict `grc netstat -tulpen` to see what ports are open when T-Pot is stopped.

**Reply by ghost6-2** [reply]
Hi there, thanks for the quick reply. Prior to realizing that my docker containers are restarting, my initial first issue was that I didnt had access to Kibana, Elastic and Spiderfoot ( 502 ERROR) and then realised my containers were restarting all the time.

I am providing screenshots below:
![yml](https://user-images.githubusercontent.com/36500152/157638861-53a41e85-08ce-49fa-add8-6105d216f96a.png)
![tulpen](https://user-images.githubusercontent.com/36500152/157638890-4c2ef585-5703-4fcf-b2e4-64ad2c82e1f5.png)

**Comment by t3chn0m4g3** [comment, marked as answer]
Please monitor the log messages when the services are starting.
While I do not see a possible conflicting port I'd recommend you to use the [netiso image](http://ftp.debian.org/debian/dists/buster/main/installer-amd64/current/images/netboot/mini.iso).

**Reply by ghost6-2** [reply]
Will try with the netiso image that you suggeested and start over. Will keep you posted about the outcome.

**Reply by t3chn0m4g3** [reply]
When asked, do only install SSH, nothing else.

**Reply by ghost6-2** [reply]
> Please monitor the log messages when the services are starting. While I do not see a possible conflicting port I'd recommend you to use the [netiso image](http://ftp.debian.org/debian/dists/buster/main/installer-amd64/current/images/netboot/mini.iso).

I installed the netiso image, and it worked like a charm! 
Thanks for the help.

Best regards.
